_posts/2024/2024-02-06-our-process.md

```diff
@@ -0,0 +1,453 @@
+> for *every* incident
+> so that there is a single point of reference for the conversation.
+> Doing this saves people from having to hunt through GitHub, a wiki,
+> and various Slack channels and/or Google docs
+> when there's a hand-off
+> or the issue occurs again.
+
+### Fixit Sprints
+
+We work in two-weeks sprints,
+and for the last year and a half,
+we've dedicated one sprint each quarter to paying down technical debt.
+If we need to clean up the data access layer in an application
+or migrate to a backward-incompatible version of a key library,
+we put that work in "our" sprint
+so that it won't disrupt work on user-facing changes.
+The benefits are hard to quantify but very real,
+and making this a regular activity makes the time easier to defend.
+
+-   Summary: 1-2 week dedicated period to address technical debt
+-   Frequency: quarterly
+-   Responsible: VP Engineering
+-   Contributors: entire engineering team
+-   Products:
+    -   Less technical debt
+
+## Team Activities
+
+Everyone on the engineering team is involved in the activities described above.
+In contrast,
+the activities below are undertaken by specific people or groups of people.
+
+### Problem Discovery
+
+If you've never had the pleasure of working with a good product manager,
+you may not realize how much better your life could be.
+This activity and the one that follows are the core of what they do:
+question, observe, and listen to users,
+figure out what their *actual* needs are,
+and translate those needs into things for engineers to fix or build.
+
+-   Summary: Ongoing investigation of users' needs
+-   Key Questions:
+    -   Who are our primary stakeholders?
+    -   What projects and iniatives are they working on?
+    -   What would accelerate their work?
+    -   Which of these needs is most pressing to solve?
+-   Frequency: ongoing
+-   Responsible: Product Manager
+-   Contributors: technical leads and selected users
+-   Products:
+    -   Prioritized lists of problems
+    -   Prioritized lists of features or fixes that would solve them
+
+### Solution Discovery
+
+This activity is where product management and engineering design overlap.
+Its output is the starting point for the design docs mentioned earlier:
+in particular,
+this is where needs, constraints, and big-picture alternatives are formulated.
+
+-   Summary: Exploration and validation of potential solutions to prioritized problems
+-   Key Questions:
+    -   Should we build it or buy it?
+    -   How could we solve this problem for our users?
+    -   Which of these solutions seems most promising and are worth further exploration?
+    -   What risks and assumptions are inherent to these solutions?
+    -   What is the fastest, cheapest way to validate these solutions?
+-   Frequency: ongoing
+-   Responsible: Product Manager and technical leads
+-   Contributors: users and engineering team members
+-   Products:
+    -   Designs for new features or applications
```

Comment:
Slight tweak: Validated designs or solutions for new features or applications.

Reply:
reworded